pkg/storage/replica_raftstorage.go

```diff
@@ -350,6 +361,7 @@ func (r *Replica) GetSnapshot(ctx context.Context) (*OutgoingSnapshot, error) {
 			return snap, err
 		}
 	}
+	return nil, errors.New("retries exhausted")
```

Comment:
I made it unlimited because the old code was unlimited too, but I see we have a context so we can use that instead, so it can be canceled without a fixed number of iterations.